src/chainbin/nnet3-chain-compute-post.cc

```diff
@@ -201,12 +201,23 @@ int main(int argc, char *argv[]) {
       // swap.
       CuMatrix<BaseFloat> gpu_nnet_output;
       gpu_nnet_output.Swap(&matrix);
+      bool all_boundary = (RandInt(0, 1) == 0);
```

Comment:
This all_boundary stuff and setting it to random was used for test.
In this case they are real boundaries, so the matrix should be (1, 0, 1, 0), but set it as:
```
boundary_mask(0, 0) = 1.0;
boundary_mask(1, 0) = 0.0;
```
and so on